Source/LinqToDB.EntityFrameworkCore/EFCoreMetadataReader.cs

```diff
@@ -600,19 +599,19 @@ string PrepareExpressionText(Expression? expr)
 				{
 					// Handling NpgSql's PostgresBinaryExpression
 
-					var left  = newExpression.GetType().GetProperty("Left")?.GetValue(newExpression) as Expression;
-					var right = newExpression.GetType().GetProperty("Right")?.GetValue(newExpression) as Expression;
+					var left  = (Expression)newExpression.GetType().GetProperty("Left")?.GetValue(newExpression)!;

```

Comment:
Exception? I don't think so.

Reply:
Expression

Reply:
I mean this code will throw exception.

Reply:
`class SqlExpression: Expression {...}`

https://github.com/npgsql/efcore.pg/blob/main/src/EFCore.PG/Query/Expressions/Internal/PostgresBinaryExpression.cs#L41-L45